src/components/ExerciseRoutine/WeekHistory/WeekHistory.tsx

```diff
@@ -7,47 +7,20 @@ const { s_calendar } = style;
 const DAYS = ['일', '월', '화', '수', '목', '금', '토'];
 
 const WeekHistory = () => {
-  const [lastDate, setLastDate] = useState(0);
-
   const today = new Date();
   const currentYear = today.getFullYear();
+  const currentMonth = today.getMonth();
   const currentDate = today.getDate();
   const currentDay = today.getDay();
-  const currentMonth = today.getMonth();
-  const weekStartDate = currentDate - currentDay;
-
-  useEffect(() => {
-    switch (currentMonth) {
-      case 0:
-      case 2:
-      case 4:
-      case 6:
-      case 7:
-      case 9:
-      case 11:
-        setLastDate(31);
-        break;
-      case 3:
-      case 5:
-      case 8:
-      case 10:
-        setLastDate(30);
-        break;
-      case 1:
-        setLastDate(new Date(currentYear, currentMonth + 1, 0).getDate());
-        break;
-      default:
-    }
-  }, [currentMonth, currentYear]);
 
   const currentWeek = useMemo(
     () =>
-      new Array(7).fill(0).map((_, index) => {
-        return weekStartDate + index > lastDate
-          ? weekStartDate + index - lastDate
-          : weekStartDate + index;
-      }),
-    [weekStartDate, lastDate]
+      new Array(7)
+        .fill(0)
+        .map((_, day) =>
+          new Date(currentYear, currentMonth, currentDate - currentDay + day).getDate()
+        ),
```

Comment:
```suggestion
        .map((_, dayOfWeek) => {
          const diffDate = currentDay - dayOfWeek;
          const dateOfWeek = currentDate - diffDate;
          return new Date(currentYear, currentMonth, dateOfWeek).getDate();
        }),
```
`currentDate - currentDay + day` 이 부분 네이밍이 좀 더 명시적이면 좋을 것 같습니다. 위에는 예시로 생각해봤는데 어떤가요?